neotest/main.d

```diff
@@ -159,6 +161,69 @@ class Get : DhtTest
     }
 }
 
+class Update : DhtTest
+{
+    import core.thread : Thread;
+    import ocean.core.array.Mutation : copy;
+
+    /// When the record has been received, pause 5s before sending back the
+    /// updated value. (This is useful for testing what happens when 2 clients
+    /// update the same record at once.)
+    bool pause;
+
+    override protected void go ( )
+    {
+        this.dht.neo.update("test".dup, 0, &this.notifier);
+        this.suspend();
+    }
+
+    private void notifier ( DhtClient.Neo.Update.Notification info,
+        Const!(DhtClient.Neo.Update.Args) args )
+    {
+        with ( info.Active ) final switch ( info.active )
+        {
+            case received:
+                auto received_record = info.received.value;
+                (*info.received.updated_value).copy(received_record);
+                (*info.received.updated_value) ~= [cast(ubyte)'X'];
+
+                if ( this.pause )
+                {
+                    Stdout.formatln("Pausing 5s");
+                    version (D_Version2)
+                        Thread.sleep(dur!("seconds")(5));
+                    else
+                        Thread.sleep(5);
```

Comment:
https://github.com/sociomantic-tsunami/ocean/blob/ac3004791f4d37669465c63fb9cc6fc09cfd1e0e/src/ocean/core/Time.d#L25

Reply:
:wave: 

Ahh! Thanks! I thought I'd seen something like that somewhere, but assumed it would be in `transition.d`.

Reply:
Updated.